src/model/test/ShadingControl_GTest.cpp

```diff
@@ -117,4 +118,125 @@ TEST_F(ModelFixture, ShadingControl_PurgeUnusedModelObjects) {
+  Model model;
+
+  Blind blind(model);
+  ShadingControl shadingControl(blind);
+
+  std::vector<Point3d> vertices1;
+  vertices1.push_back(Point3d(0,0,1));
+  vertices1.push_back(Point3d(0,0,0));
+  vertices1.push_back(Point3d(1,0,0));
+  vertices1.push_back(Point3d(1,0,1));
+  SubSurface subSurface1(vertices1, model);
+
+  std::vector<Point3d> vertices2;
+  vertices2.push_back(Point3d(0,0,1));
+  vertices2.push_back(Point3d(0,0,0));
+  vertices2.push_back(Point3d(1,0,0));
+  vertices2.push_back(Point3d(1,0,1));
+  SubSurface subSurface2(vertices2, model);
+
+  shadingControl.setSubSurfaces({subSurface1, subSurface2});
+  EXPECT_EQ(2, shadingControl.numberofSubSurfaces());
+
+  // Clone same model
+  {
+    // TODO: what is the expected behavior here? Should the clone reference the same subSurfaces as the original?
+    ShadingControl shadingControlClone = shadingControl.clone(model).cast<ShadingControl>();
+    ASSERT_TRUE(shadingControlClone.shadingMaterial());
+    EXPECT_EQ(shadingControl.shadingMaterial()->handle(), shadingControlClone.shadingMaterial()->handle());
+    EXPECT_EQ(2u, shadingControl.numberofSubSurfaces());
+    EXPECT_EQ(2u, shadingControl.numExtensibleGroups());
+    EXPECT_EQ(2u, shadingControlClone.numberofSubSurfaces());
+    EXPECT_EQ(2u, shadingControlClone.numExtensibleGroups());
+    EXPECT_EQ(shadingControl.subSurfaces(), shadingControlClone.subSurfaces());
+
+  }
+
+  // Clone other model
+  {
+    Model model2;
+
+    // TODO: what is the expected behavior here? It will clone the Construction and ShadingMaterial (ResourceObject...) referenced here, but not the subSurfaces
+    ShadingControl shadingControlClone = shadingControl.clone(model2).cast<ShadingControl>();
+    EXPECT_EQ(2u, shadingControl.numberofSubSurfaces());
+    EXPECT_EQ(2u, shadingControl.numExtensibleGroups());
+    EXPECT_EQ(0u, shadingControlClone.numberofSubSurfaces());
+    EXPECT_EQ(0u, shadingControlClone.numExtensibleGroups());
+    EXPECT_EQ(1u, model2.getConcreteModelObjects<Blind>().size());
+    EXPECT_TRUE(shadingControlClone.shadingMaterial());
+    EXPECT_NE(shadingControl.shadingMaterial().get(), shadingControlClone.shadingMaterial().get());
+  }
+}
+
+TEST_F(ModelFixture, ShadingControl_RemoveRequiredObject) {
+  Model model;
+
+  Blind blind(model);
+  ShadingControl shadingControl(blind);
+
+  blind.remove();
+
+  EXPECT_FALSE(shadingControl.shadingMaterial());
+  EXPECT_FALSE(shadingControl.construction());
+
+  // TODO: You're now in a broken, and unfixable state: you neither have a Construction nor a ShadingMaterial,
+  // and the model API has no setters to help you fix the state (setShadingMaterial / setConstruction)
+  // This WILL get forward translated anyways. We need to determine whether we want to add setters in the Model API or have the FT do a check and not
+  // translate the object if it doesn't have either. Throwing here so we do not forget to do it
+  ASSERT_TRUE(false);
+}
```

Comment:
TODO:

* [ ] Decide what to do with this. I would say having the FT not translate the object if it's invalid is a quick and helpful fix

Reply:
(Note: this is a pre-existing condition, not something related to this PR... I just noticed it while testing things)